modules/kernel/src/main/scala-2.11/notebook/ReplCalculator.scala

```diff
@@ -429,8 +433,9 @@ class ReplCalculator(
           thisSender ! ErrorResponse("Incomplete (hint: check the parenthesis)", incomplete = true)
       }
 
-      result onComplete {
-        _ => thisSelf ! "process-next"
+      result onComplete { _ =>
+        currentlyExecutingTask = None
```

Comment:
P.S. i don't quite like that this line is executed inside of `future.onComplete`.
thus I'm afraid `currentlyExecutingTask = None` should go into `onFailure`.

certainly there's still something stinky about this way of pipelining... let's think how to make it more clean